### ISSUE TITLE
Add Widevine support

### ISSUE THREAD

**Issue by sentialx** [wexond/browser-base#175]
This will allow to play songs on Spotify and videos on Netflix. I've sent an email to Google about Widevine plugin but they didn't seem to respond. #174

**Comment by torstenlemke** [comment]
Hey there, I have been following the issue with the Widevine encryption thingy for a while now ... does this maybe help?: https://electronjs.org/docs/tutorial/testing-widevine-cdm

**Comment by sentialx** [comment]
Yeah, but I have to get it somewhere without licensing issues. Google doesn't want to publish the plugin for everyone.

**Comment by torstenlemke** [comment]
Hmm ... I don't get google's policies about such stuff ...

**Comment by bsak2003** [comment]
> Yeah, but I have to get it somewhere without licensing issues. Google doesn't want to publish the plugin for everyone.

Any progress with Widevine? 

PS: I suggest you using an older version of Widevine CDM (like a Firefox, which uses 4.10.1146.0 instead of latest 4.10.1196.0 used by Chrome/Opera) because of the latest version drop frames in some cases. But anyway, you need a license from Google in the first place :C

**Comment by sentialx** [comment]
What comes to my mind is Wexond could download widevine from Google. As I can see it's legal.

**Comment by bsak2003** [comment]
> What comes to my mind is Wexond could download widevine from Google. As I can see it's legal.

I think you will only need a certification when Wexond will be finished. But I am not sure :) 

and you're thinking about downloading Widevine CDM (excatly files widevinecdm.dll & widevinecdm.dll.sig) directly from Google? I'm sure that Mozilla Firefox does the same because when I open netflix.com on FF, I got "Firefox is downloading Google Widevine CDM" notification.

**Comment by samuelmaddock** [comment]
@sentialx You'll want to read electron/electron#12427 and [my blog post](https://blog.samuelmaddock.com/posts/google-widevine-blocked-my-browser/) about the response I received. [Verified Media Path (VMP)](https://www.widevine.com/news) will soon be an enforced requirement which will prevent browsers from simply including the Widevine binaries.

**Comment by sentialx** [comment]
@samuelmaddock So what are the possible solutions as for now? I've wrote an email to widevine-support@google.com this time instead of widevine@google.com.

**Comment by sentialx** [comment]
Update, their bot responded to me with this email

```
Hello,

It appears you have attempted to contact this alias directly. All direct emails to this alias will not be responded to. Instead, please visit us below to find the appropriate contact form:

http://www.widevine.com/contact.html

Thank you,
The Google Widevine Team
```
But the link to their contact form doesn't work either. 

EDIT: Here I've found the updated contact form: https://support.google.com/widevine/troubleshooter/6027072

**Comment by samuelmaddock** [comment]
The only solution I know of is to somehow complete the agreement with Widevine. This may require incorporating a company and/or a closed source project, I'm not entirely sure as they weren't clear in their responses to me.

I'm abandoning Electron in my own project and moving to a web app with a required browser extension.

**Comment by bsak2003** [comment]
> The only solution I know of is to somehow complete the agreement with Widevine. This may require incorporating a company and/or a closed source project, I'm not entirely sure as they weren't clear in their responses to me.
> 
> I'm abandoning Electron in my own project and moving to a web app with a required browser extension.

So how Mozilla include Widevine CDM into a Firefox? I've checked and Firefox automaticaly downloads Widevine when I'm opening open.spotify.com or netflix.com - even on self-compiled (from their Mercurial repo) browser.

**Comment by tongdu99** [comment]
Brave browser can prompt to install widevine extension. After unblock the guard, it can play netflix video successfully. So it seems like it should work for other open source browser.

**Comment by sentialx** [comment]
Currently Widevine doesn't work in the newest versions of Electron: https://github.com/electron/electron/issues/20211

Maybe somewhere in the future I will be finally able to obtain the license from Google...

**Comment by sentialx** [comment]
I think we could just prompt to install widevine as @tongdu99 says.

**Comment by radicaldrew** [comment]
Not sure if this is still relevant but I was able to get Widevine support working including Netflix by using a custom electron build.  I could either provide a pull request or instructions.

**Comment by oscartbeaumont** [comment]
@radicaldrew If you got it working it would be awesome to see how you did it as I have a similar problem to this one with my app [ElectronPlayer](https://github.com/oscartbeaumont/ElectronPlayer) which is similarly based on Electron.
I have it working in Linux using the [Castlabs electron fork](https://github.com/castlabs/electron-releases) with the default Widevine development certificate but am unable to get anything Electron-based to work on Mac although my other project [NativeKit](https://github.com/oscartbeaumont/NativeKit) is looking like the next best option if I can find the time to continue its development.

**Comment by radicaldrew** [comment]
I have actually switched to the castlabs electron now and it already has the widevine certs built in.
Unless you can figure a way to check if the OS has chrome installed and copy and past the files
this seems like the best option.

**Comment by tarob0ba** [comment]
This is late (sorry for the bump), but I have been able to get Widevine working in Wexond (Netflix and Spotify work!) by using the castlabs fork of Electron and their EVS signing, which is free and takes two minutes to set up. I'm happy to share details if that helps.

**Comment by jtpotato** [comment]
> Not sure if this is still relevant but I was able to get Widevine support working including Netflix by using a custom electron build. I could either provide a pull request or instructions.

@radicaldrew

Instructions would be useful, was planning on forking the electron version of wexond to continue development.
